packages/FormElement/src/components/Label/Label.js

```diff
@@ -4,56 +4,70 @@ import PropTypes from "prop-types";
 import useI18n from "@paprika/l10n/lib/useI18n";
 
 import labelStyles, { ruleStyles } from "./Label.styles";
+import Help from "../Help/Help";
+import { FormElementContext } from "../../FormElement";
 
 const propTypes = {
+  children: PropTypes.node.isRequired,
   /** If "optional" text should be displayed beside the label  */
-  hasOptionalLabel: PropTypes.bool.isRequired,
+  hasOptionalLabel: PropTypes.bool,
   /** If "require" text should be displayed beside the label */
-  hasRequiredLabel: PropTypes.bool.isRequired,
+  hasRequiredLabel: PropTypes.bool,
   /** Help indicator */
   help: PropTypes.node,
   /** id for the element */
   id: PropTypes.string,
-  /** Should label and children be inline or not */
-  isInline: PropTypes.bool.isRequired,
   /** Should label be hidden */
-  isVisuallyHidden: PropTypes.bool.isRequired,
-  /** Label text of the input field */
-  label: PropTypes.node.isRequired,
-  /** Set if FormElement contains multiple children to render a legend element instead of label */
-  hasFieldSet: PropTypes.bool.isRequired,
+  isVisuallyHidden: PropTypes.bool,
 
   onClick: PropTypes.func,
 };
 
 const defaultProps = {
+  hasOptionalLabel: false,
+  hasRequiredLabel: false,
   help: null,
   id: null,
+  isVisuallyHidden: false,
   onClick: () => {},
 };
 
-const Label = React.forwardRef((props, ref) => {
-  const { hasOptionalLabel, hasRequiredLabel, help, id, hasFieldSet, label, ...moreProps } = props;
+const Label = props => {
+  const { hasFieldSet, formElementA11yProps } = React.useContext(FormElementContext);
+  const { hasOptionalLabel, hasRequiredLabel, help, id, children, ...moreProps } = props;
 
   const I18n = useI18n();
 
-  const labelProps = hasFieldSet ? { as: "legend" } : { as: "label", htmlFor: id, ref };
+  const labelProps = hasFieldSet ? { as: "legend" } : { as: "label" };
+
+  const a11yProps = formElementA11yProps
+    ? { htmlFor: formElementA11yProps.labelA11yProps.id, ref: formElementA11yProps.labelA11yProps.ref }
+    : {};
+
+  const renderQuestionRequirement = () => {
+    if (hasRequiredLabel) {
+      return I18n.t("formElement.required");
+    }
+    if (hasOptionalLabel) {
```

Comment:
For now i'm leaving in support for the "Optional" tag to be shown with the formElement, although i'm not sure if we really need or want it anymore. Perhaps designers can confirm @sunminsarahlee

Reply:
Last I saw, both are still supported in the Forms patterns pitch:
https://aclgrc.atlassian.net/wiki/spaces/UX/pages/676398749/Forms+pattern+v2+in+review#Element-instruction%3A-Concise